enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,120 @@
+---
+title: sdk-metrics
+  - "@varshaprasad96"
+reviewers:
+  - "@dmesser"
+  - "@joelanford"
+approvers:
+  - "@dmesser" 
+  - "@joelanford"
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implementable
+---
+
+# Operator SDK metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The enhancement is intended to define the bundle resources in which SDK labels can be added, so that they can be read by OLM and included in the current OLM metrics. These metrics will help us in providing a measure to evaluate the reachability of Operator SDK among community operators.
+
+## Motivation
+
+Collecting metrics regarding the usage of Operator SDK among operator authors is important to measure the reachability of the tool. Since OLM already has the pipeline built to obtain metrics on the operators and report it to the redhat telemetry, we intend to utilize the same to obtain data on the operators using SDK to build bundle images.
+
+### Goals
+
+The goal of this enhancement is to :
+1.  List the bundle resources on which SDK stamp will be added.
+2. The format of the stamp which will be added.
+
+### Design details
+
+The current bundle image has the following project structure:
+
+```
+ ...
+  bundle.Dockerfile
+  ...
+  $ tree ./deploy/olm-catalog/test-operator/
+  ./deploy/olm-catalog/test-operator/
+  ├── manifests
+  │   ├── example.com_tests_crd.yaml
+  │   └── test-operator.clusterserviceversion.yaml
+  └── metadata
+      └── annotations.yaml
+```
+
+SDK stamps would be present in the following locations:
+
+1. bundle.Dockerfile
+
+The dockerfile will include the sdk stamp in the form of `LABEL` which will follow the following format:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = operator_type:sdk_version
```

Comment:
@ecordell Is the `operators.operatorframework.io.bundle` prefix reserved for the bundle media type?

We need to coordinate what labels are safe to add to bundle images since they have special meaning for OLM.

Related to this is #18, which is seeking to add a separate media type related to scorecard metadata and image contents. Do we need to define a new media type for SDK metadata and do something like the following:

```
sdk.operatorframework.io.metadata.mediatype = "sdk+v1"
sdk.operatorframework.io.metadata.version = "v0.17.0"
sdk.operatorframework.io.metadata.layout = "go.kubebuilder.io/v2.0.0"
```

Reply:
+1 on the proposed metadata - that seems valuable and will avoid collisions

Reply:
Something to consider: if/when we split up the SDK into component repos (CLI/Go plugin, Helm, Ansible, libraries), what will we use for `sdk.operatorframework.io.metadata.version.v1`?

Because we're versioning label keys/mediatype, we can punt on this. If we decide to use a version other than that of the `operator-sdk` binary, we can create a `version.v2` label.